### PR TITLE
feat: update UML sequence diagram in developer guide to show sequence for export command

### PR DESCRIPTION
**Pull request by danielleloh**
the sequence diagram is useful in showing how the components of the address book interact with each other to respond to a user command. in this case, the user command we are showing is `export`.

changes made include,
* sequence diagram for `delete` command has been updated to a sequence diagram for `export` command, which is a new feature we included in our v1.2
* developer guide has been updated to change any information related to previous sequence diagram to be relevant to the current updated sequence diagram

**other note,**
* imported packages and their classes which are used in ExportCommand were not included in the sequence diagram. Some input on whether it is relevant and should be added into the diagram would be appreciated :)

* edit: included delete sequence diagram in both diagrams and developer guide

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -90,22 +90,23 @@ Here's a (partial) class diagram of the `Logic` component:
 
 <puml src="diagrams/LogicClassDiagram.puml" width="550"/>
 
-The sequence diagram below illustrates the interactions within the `Logic` component, taking `execute("delete 1")` API call as an example.
+The sequence diagram below illustrates the interactions within the `Logic` component, taking `execute("export")` API call as an example.
 
-<puml src="diagrams/DeleteSequenceDiagram.puml" alt="Interactions Inside the Logic Component for the `delete 1` Command" />
```
Comment (delishad21) [inline docs/DeveloperGuide.md:95]: Actually maybe we can consider just leaving both diagrams in?

Comment (delishad21) [reply docs/DeveloperGuide.md:95]: Especially since export dosent go thru the parser process as well

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -90,22 +90,23 @@ Here's a (partial) class diagram of the `Logic` component:
...
+<puml src="diagrams/ExportSequenceDiagram.puml" alt="Interactions Inside the Logic Component for the `export` Command" />
+
+How the `Logic` component works:
+
+1. When `Logic` is called upon to execute a command, it is passed to an `AddressBookParser` object which in turn creates a parser that matches the command (e.g., `DeleteCommandParser`) and uses it to parse the command.
 
 <box type="info" seamless>
 
-**Note:** The lifeline for `DeleteCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline continues till the end of diagram.
+**Note:** For the `export` command above, a parser that matches the command is not created by `AddressBookParser` as it only contains a command and does not contain an argument. However, for other relevant commands, `AddressBookParser` will create the parser that matches the command.
```
Comment (delishad21) [inline docs/DeveloperGuide.md:103]: Good clarification on the export command diagram not using an export command parser

**Review by delishad21 — COMMENTED** [review]
Just a small suggestion to leave the original delete diagram in as well. Looks good to me otherwise, nice job on the diagram!

File: docs/diagrams/ExportSequenceDiagram.puml
Comment (jayllo-c) [inline docs/diagrams/ExportSequenceDiagram.puml:1]: great job on adding the sequence diagram for the export command!

**Review by Pughal77 — COMMENTED** [review]
Looks good to me :)

**Pull request merged**